app/middleware.go

```diff
@@ -183,3 +194,48 @@ func (nr *NewRelicMiddleware) Serve(next echo.HandlerFunc) echo.HandlerFunc {
 		return nil
 	}
 }
+
+const metricName = "response_time_milliseconds"
+
+// ResponseTimeMetricsMiddleware struct encapsulating DDStatsD
+type ResponseTimeMetricsMiddleware struct {
+	DDStatsD *middleware.DogStatsD
+}
+
+//ResponseTimeMetricsMiddleware is a middleware to measure the response time
+//of a route and send it do StatsD
+func (responseTimeMiddleware ResponseTimeMetricsMiddleware) Serve(next echo.HandlerFunc) echo.HandlerFunc {
+	return func(c echo.Context) error {
+
+		metricTagsMap := make(map[string]interface{})
+		c.Set("metricTagsMap", metricTagsMap)
+
+		startTime := time.Now()
+		result := next(c)
+		status := c.Response().Status()
+		route := c.Path()
+		method := c.Request().Method()
+		caller := c.Get("metricTagsMap")
+		caller = caller.(map[string]interface{})["gameID"]
```

Comment:
Let's make this safer:

```suggestion
        var caller string
		if metricTagsMap, ok := c.Get("metricTagsMap").(map[string]interface{}); ok {
				caller = metricTagsMap["gameID"]
		}
```

Reply:
when setting caller as string type, I get the following error: `interface conversion: interface {} is nil, not string`, the same happens if I try to convert the caller final value to string.